crates/matrix-sdk-crypto/src/verification/mod.rs

```diff
@@ -83,6 +83,55 @@ pub struct Emoji {
     pub description: &'static str,
 }
 
+/// The list of emojis that are used for interactive verification using a short
+/// auth string.
+///
+/// This will contain exactly 7 emojis, including their descriptions, from the
+/// list of emojis from the [sepc].
+///
+/// [spec]: https://spec.matrix.org/unstable/client-server-api/#sas-method-emoji
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EmojiList(pub [Emoji; 7]);
+
+impl EmojiList {
+    /// Format the the list of emojis as a two line string.
+    ///
+    /// The first line will contain the emojis spread out so the second line can
+    /// contain the descriptions centered bellow the emoji.
+    pub fn format(&self) -> String {
+        let (emojis, descriptions): (Vec<_>, Vec<_>) =
+            self.0.iter().map(|e| (e.symbol, e.description)).unzip();
+
+        let center_emoji = |emoji: &str| -> String {
+            const EMOJI_WIDTH: usize = 2;
+            // These are emojis that need VARIATION-SELECTOR-16 (U+FE0F) so that they are
+            // rendered with coloured glyphs. For these, we need to add an extra
+            // space after them so that they are rendered properly in terminals.
+            const VARIATION_SELECTOR_EMOJIS: [&str; 7] = ["☁️", "❤️", "☂️", "✏️", "✂️", "☎️", "✈️"];
+
+            // Hack to make terminals behave properly when one of the above is printed.
+            let emoji = if VARIATION_SELECTOR_EMOJIS.contains(&emoji) {
+                format!("{} ", emoji)
+            } else {
+                emoji.to_string()
+            };
+
+            // This is a trick to account for the fact that emojis are wider than other
+            // monospace characters.
```

Comment:
I'm pretty sure how wide emojis are rendered in monospace-expecting places depends entirely on the app, but maybe this is a good default 🤷🏼

Reply:
It does depend on the app, since this was used for quite a while in weechat-matrix it seems to be a good enough default.